tests/Monolog/Handler/StreamHandlerTest.php

```diff
@@ -184,33 +184,40 @@ public function testWriteNonExistingFileResource()
     /**
      * @covers Monolog\Handler\StreamHandler::__construct
      * @covers Monolog\Handler\StreamHandler::write
+     * @dataProvider provideNonExistingAndNotCreatablePath
      */
-    public function testWriteNonExistingAndNotCreatablePath()
+    public function testWriteNonExistingAndNotCreatablePath($nonExistingAndNotCreatablePath)
     {
-        $this->expectException(\Exception::class);
-        $this->expectExceptionMessage('There is no existing directory at');
-
-
         if (defined('PHP_WINDOWS_VERSION_BUILD')) {
             $this->markTestSkipped('Permissions checks can not run on windows');
         }
-        $handler = new StreamHandler('/foo/bar/'.rand(0, 10000).DIRECTORY_SEPARATOR.rand(0, 10000));
-        $handler->handle($this->getRecord());
-    }
 
-    /**
-     * @covers Monolog\Handler\StreamHandler::__construct
-     * @covers Monolog\Handler\StreamHandler::write
-     */
-    public function testWriteNonExistingAndNotCreatableFileResource()
-    {
+        $handler = null;
+
+        try {
+            $handler = new StreamHandler($nonExistingAndNotCreatablePath);
+        } catch (\Exception $fail) {
+            $this->fail(
+                'A non-existing and not creatable path should throw an Exception earliest on first write.
+                 Not during instantiation.'
+            );
+        }
+
         $this->expectException(\Exception::class);
```

Comment:
This could be changed to `\UnexpectedValueException::class` I believe?

Reply:
Sorry, I somehow missed your review. I just committed an update, that expects an `\UnexpectedValueException::class`.